File: macos/QMK Toolbox/USB/Bootloader/BootloadHIDDevice.m
#import "BootloadHIDDevice.h"

@implementation BootloadHIDDevice

- (id)initWithUSBDevice:(USBDevice *)usbDevice {
    if (self = [super initWithUSBDevice:usbDevice]) {
        self.name = @"BootloadHID";
        self.type = BootloaderTypeBootloadHID;
        self.resettable = YES;
    }
    return self;
}

-(void)flashWithMCU:(NSString *)mcu file:(NSString *)file {
    [self runProcess:@"bootloadHID" withArgs:@[@"-r", file]];
}

-(void)resetWithMCU:(NSString *)mcu {
    [self runProcess:@"bootloadHID" withArgs:@[@"-r"]];
}

@end

File: macos/QMK Toolbox/USB/Bootloader/BootloadHIDDevice.swift
import Foundation

class BootloadHIDDevice: BootloaderDevice {
    override init(usbDevice: USBDevice) {
        super.init(usbDevice: usbDevice)
        name = "BootloadHID"
        type = .bootloadHid
        resettable = true
    }

    override func flash(_ mcu: String, file: String) {
        runProcess("bootloadHID", args: ["-r", file])
    }

    override func reset(_ mcu: String) {
        runProcess("bootloadHID", args: ["-r"])
    }
}
